pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -147,11 +147,19 @@ func parseConfig(getconfigCtx *getconfigContext, config *zconfig.EdgeDevConfig,
 		parseSystemAdapterConfig(getconfigCtx, config, source, forceSystemAdaptersParse)
 
 		if source != fromBootstrap {
-			parseBaseOS(getconfigCtx, config)
+			activateNewBaseOSFlag := parseBaseOS(getconfigCtx, config)
 			parseNetworkInstanceConfig(getconfigCtx, config)
 			parseContentInfoConfig(getconfigCtx, config)
 			parseVolumeConfig(getconfigCtx, config)
 
+			if source == fromController && activateNewBaseOSFlag {
+				// We need to activate the new baseOS
+				// before we can process the app instances
+				// which depend on the new baseOS
+				log.Noticef("parseConfig: Ignoring config as a new baseOS image is being activated")
+				return skipConfig
```

Comment:
Given that you run through this loop every 60 seconds or so, this logic (even with ParseBaseOs fixed per below) will only capture the first transition of Activate becoming true for a BaseOsConfig. We also need to remember this state.

Potentially one can do that by returning true here (which will record true in ctx.configProcessingSkipFlag) but that will skip more than it should on the next call. So I think it makes sense to introduce a seond skip flag as a return value and record that in a second place in ctx. We could rename the existing configProcessingSkipFlagReboot and the new one configProcessingSkipFlagUpdate.

Reply:
I've split the flag into two. I hope I understood the logic right and it handles the cases properly now...